src/ethereum_spec_tools/sync.py

```diff
@@ -179,70 +189,54 @@ def __init__(self) -> None:
         self.log = logging.getLogger(__name__)
         self.options = self.parse_arguments()
 
-        database_path = None
-
-        if self.options.persist is not None:
-            database_path = os.path.join(self.options.persist, "state")
-
-            try:
-                os.mkdir(self.options.persist)
-            except FileExistsError:
-                pass
-
-        if self.options.optimized:
-            import ethereum_optimized
-
-            ethereum_optimized.monkey_patch(state_path=database_path)
-
         self.forks = Hardfork.discover()
         self.active_fork_index = 0
 
-        if self.options.persist is None:
-            self.chain = self.module("spec").apply_fork(None)
-            return
-
-        self.log.debug("loading blocks and state...")
-
-        blocks_path = os.path.join(self.options.persist, "blocks.pickle")
-
-        blocks = []
+        if self.options.optimized:
+            import ethereum_optimized
 
-        try:
-            with open(blocks_path, "rb") as f:
-                while True:
-                    try:
-                        blocks.extend(pickle.load(f))
-                    except EOFError:
-                        break
-        except FileNotFoundError as e:
-            self.log.warning("no block file found", exc_info=e)
-            self.chain = self.module("spec").apply_fork(None)
-            return
+            ethereum_optimized.monkey_patch(state_path=self.options.persist)
+        else:
+            if self.options.persist is not None:
+                self.log.error("--perist is not supported without --optimized")
+                exit()
 
-        if len(blocks) == 0:
-            raise Exception("no blocks loaded")
+        state = self.module("state").State()
 
-        # TODO: Replace self.chain.state with the correct hard fork.
+        if self.options.persist is not None:
+            persisted_block = self.active_fork.optimized_module(
+                "state_db"
+            ).get_metadata(state, b"block_number")
 
-        if self.options.optimized:
-            state = self.active_fork.optimized_module("state_db").State()
+            if persisted_block is not None:
+                persisted_block = int(persisted_block)
         else:
-            state_path = os.path.join(self.options.persist, "state.pickle")
+            persisted_block = None
 
-            try:
-                with open(state_path, "rb") as f:
-                    state = pickle.load(f)
-            except FileNotFoundError as e:
-                raise Exception("found blocks file but no state") from e
-
-        self.chain = self.module("spec").BlockChain(
-            blocks=blocks,
-            state=state,
-        )
-
-        # TODO: Fast forward to correct hard fork.
+        if persisted_block is not None:
```

Comment:
Would you mind flipping the order of these conditions (so `if persisted_block is None` comes first)? I like avoiding extra negation if possible.